### DISCUSSION TITLE
Just saying thanks

### DISCUSSION THREAD

**Discussion by impredicative** [benbrandt/text-splitter#138] [Show and tell]
Thanks for this package! I am now [using](https://github.com/impredicative/podgenai/blob/master/src/podgenai/util/semantic_text_splitter.py) it in a personal project.

**Comment by benbrandt** [comment]
Awesome, thanks @impredicative ! Makes me very happy to hear. 

I've been trying to capture upcoming feature ideas here: https://github.com/users/benbrandt/projects/2
Let me know if you run into any issues, or have ideas that I am missing that would help you out. Best of luck with the project!

**Comment by impredicative** [comment]
My personal interest is in the continued correctness of the output chunks. If the package becomes more complicated with more features, there is a hypothetical risk of an error being introduced due to an edge case, adversely affecting even the existing usage. Perhaps a good test suite will guard against this risk, and forgive me if it already exists.

By correctness, I mean:
1. The chunks, if they were to be merged back, is always identical to the original text. This is of primary importance.
2. The max character limit is always ensured. This too is of primary importance.
3. The split points are exactly where expected. This is of secondary importance.

**Reply by benbrandt** [reply]
Yep makes sense. This is definitely something I want to guard against as well. 

Regarding 1: this is true if trimming isn't used. If trimming is on, there will be missing whitespace. But there are tests for this for sure.

Regarding 2: This is also tested, and certainly very important

Regarding 3: This is difficult for sure, as "expected" can be subjective, but I try my best :)
